backend/book-app/src/main/java/com/karankumar/bookproject/controller/UserController.java

```diff
@@ -165,15 +165,32 @@ public void updateEmail(@RequestParam("newEmail") String newEmail,
     @ResponseStatus(HttpStatus.OK)
     public boolean updatePassword(@RequestParam("currentPassword") String currentPassword,
                                @RequestParam("newPassword") String newPassword) throws MessagingException {
-        User user = userService.getCurrentUser();
 
-        if (passwordEncoder.matches(currentPassword, user.getPassword())) {
-            userService.changeUserPassword(user, newPassword);
-            emailService.sendMessageUsingThymeleafTemplate(
-                    user.getEmail(),
-                    EmailConstant.ACCOUNT_PASSWORD_CHANGED_SUBJECT,
-                    EmailTemplate.getChangePasswordEmailTemplate(emailService.getUsernameFromEmail(user.getEmail()))
+        User user;
+
+        try {
+            user = userService.getCurrentUser();
+        } catch (com.karankumar.bookproject.backend.service.CurrentUserNotFoundException ex) {
+            throw new ResponseStatusException(
+                    HttpStatus.NOT_FOUND,
+                    CURRENT_USER_NOT_FOUND_ERROR_MESSAGE
             );
+        }
+
+        if (passwordEncoder.matches(currentPassword, user.getPassword())) {
+            try{
+                userService.changeUserPassword(user, newPassword);
```

Comment:
It would be good to check password strength and then proceed with the password change.
The password strength could be the criteria to check if the password is valid.